src/module_compilation.ml

```diff
@@ -78,7 +78,7 @@ let build_cm sctx ?sandbox ~dynlink ~flags ~cm_kind ~(dep_graph:Ocamldep.dep_gra
          Build.dyn_paths (Build.arr (lib_dependencies ~cm_kind)) >>>
          Build.run ~context:ctx (Dep compiler)
            ~extra_targets
-           [ Ocaml_flags.get_for_cm flags ~cm_kind
+           [ Ocaml_flags.get_for_cm flags ~target:m.impl.name ~cm_kind
```

Comment:
BTW, the argument shouldn't be named `target`, since we are matching on the source name